src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -263,6 +267,69 @@ public OrganisationsDetailResponse retrieveOrganisations() {
         return updatedOrganisationDetails;
     }
 
+    public ResponseEntity retrieveActiveOrganisationDetails(Pageable pageable) {
+
+        List<Organisation> updatedOrganisationDetails = new ArrayList<>();
+        Map<String,Organisation> activeOrganisationDtls = new ConcurrentHashMap<String,Organisation>();
+        OrganisationsDetailResponse organisationsDetailResponse = null;
+
+        Page<Organisation> activeOrganisations = organisationRepository.findByStatus(
+                OrganisationStatus.ACTIVE, pageable);
+        List<Organisation> activeOrganisationsTemp = activeOrganisations.getContent();
+        if (CollectionUtils.isEmpty(activeOrganisationsTemp)) {
+            throw new EmptyResultDataAccessException(1);
+        } else {
+            activeOrganisationsTemp.forEach(organisation -> {
+                if (organisation.getUsers().size() > 0 && null != organisation.getUsers().get(0).getUserIdentifier()) {
+                    activeOrganisationDtls.put(organisation.getUsers().get(0).getUserIdentifier(),organisation);
+                }
+            });
+        }
+
+        if (!CollectionUtils.isEmpty(activeOrganisationDtls)) {
+            RetrieveUserProfilesRequest retrieveUserProfilesRequest = new RetrieveUserProfilesRequest(activeOrganisationDtls.keySet().stream().sorted().collect(Collectors.toList()));
+            updatedOrganisationDetails = RefDataUtil.getMultipleUserProfilesFromUp(userProfileFeignClient,retrieveUserProfilesRequest,
+                    "false", activeOrganisationDtls);
+            organisationsDetailResponse = new OrganisationsDetailResponse(
+                    updatedOrganisationDetails, true);
+        }
+
+        HttpHeaders headers = RefDataUtil.generateResponseEntityWithPaginationHeader(pageable, activeOrganisations, null);
+        return ResponseEntity.status(200).headers(headers).body(organisationsDetailResponse);
+    }
+
+    public List<Organisation> retrieveActiveOrganisationDetails(List<Organisation> organisations) {
+
+        List<Organisation> updatedOrganisationDetails = new ArrayList<>();
+        Map<String,Organisation> activeOrganisationDtls = new ConcurrentHashMap<String,Organisation>();
+        List<Organisation> pendingOrganisations = new ArrayList<>();
+        List<Organisation> activeOrganisations = new ArrayList<>();
+
+        if (!organisations.isEmpty()) {
```

Comment:
Refactor into a separate class, either a helper or validator service or some combination of the two. These methods are two complex and hard-to-test, need to break this down.